lib/validate-selectors.js

```diff
@@ -2,6 +2,56 @@ var listSequences = require('./list-sequences');
 var shouldIgnoreRule = require('./should-ignore-rule');
 var shouldIgnoreSelector = require('./should-ignore-selector');
 var toInterpoloatedRegexp = require('./to-interpolated-regexp');
+var resolveNestedSelector = require('postcss-resolve-nested-selector');
+
+function isTopLevel(node) {
+  return node.type === 'root';
+}
+
+function isNestedRule(node) {
+  return /(?:at)?rule/.test(node.parent.type)
+}
+
+function hasNoDeclarations(node) {
+  return node.nodes.length && node.every(function(child) {
+    return child.type !== 'decl';
+  });
+}
+
+function getComponentRootRule(node) {
+  while (!isTopLevel(node.parent)) {
+    node = node.parent;
+  }
+  return node;
+}
+
+function unWrapSelectors(parent, rule) {
+  var selectors = [];
+  parent.walkRules(function(node) {
+    // Only unwrap as far as the current rule being linted
+    if (node.selector !== rule.selector) {return;}
+    node.selectors.forEach(function(selector) {
+      selectors = selectors.concat(resolveNestedSelector(selector, node));
+    });
+  });
+  return selectors;
+}
+
+function getSelectors(rule) {
+  // Skip validation on rules with no declarations
+  // as these don't exist after rules have been unwrapped
+  if (hasNoDeclarations(rule)) {
+    return null;
```

Comment:
A falsey value felt more appropriate otherwise I have to check `if (!selectors.length) {return;}` further down.